tiledb/sm/array_schema/domain.cc

```diff
@@ -61,6 +61,33 @@ Domain::Domain() {
   cell_num_per_tile_ = 0;
 }
 
+Domain::Domain(
+    uint64_t cell_num_per_tile,
+    Layout cell_order,
+    const std::vector<std::shared_ptr<Dimension>> dimensions,
+    Layout tile_order)
+    : cell_num_per_tile_(cell_num_per_tile)
+    , cell_order_(cell_order)
+    , dimensions_(dimensions)
+    , dim_num_((unsigned int)dimensions.size())
+    , tile_order_(tile_order) {
+  if (cell_num_per_tile_ == 0) {
```

Comment:
The test here is incorrect. `cell_num_per_tile_` is always initialized from other member variables.